googletest/include/gtest/gtest-printers.h

```diff
@@ -681,14 +679,41 @@ class UniversalPrinter {
   GTEST_DISABLE_MSC_WARNINGS_POP_()
 };
 
-#if GTEST_HAS_ABSL
+#if GTEST_INTERNAL_HAS_ANY
+
+// Printer for std::any / absl::any
 
-// Printer for absl::optional
+template <>
+class UniversalPrinter<Any> {
+ public:
+  static void Print(const Any& value, ::std::ostream* os) {
+    if (value.has_value())
+      *os << "'any' type with value of type " << GetTypeName(value);
+    else
+      *os << "'any' type with no value";
+  }
+
+ private:
+  static std::string GetTypeName(const Any& value) {
+#if GTEST_HAS_RTTI
+    return internal::GetTypeName(value.type());
+#else
+    static_cast<void>(value); // possibly unused
+    return "the element type";
```

Comment:
I don't think "the element type" is more informative than just "<type>". Maybe just call `internal::GetTypeName`?

Reply:
I based that on [VariantMatcher<T>::GetTypeName()](https://github.com/google/googletest/blob/6f5fd0d7199b9a19faa9f499ecc266e6ae0329e7/googlemock/include/gmock/gmock-matchers.h#L3605) and [AnyCastMatcher<T>::GetTypeName()](https://github.com/google/googletest/blob/6f5fd0d7199b9a19faa9f499ecc266e6ae0329e7/googlemock/include/gmock/gmock-matchers.h#L3666). It's the same on the current master too. Funny, I had the same impression as you ;)